src/GraphQL/Types/Composite/InputObjectGraphType.cs

```diff
@@ -50,12 +52,56 @@ internal InputObjectGraphType(InputObjectGraphType<TSourceType>? cloneFrom)
             : base(cloneFrom)
         {
             // if (cloneFrom == null) { /* initialization logic */ }
+
+            if (typeof(TSourceType) == typeof(object))
+            {
+                // for InputObjectGraphType just return the dictionary
+                _parseDictionary = static x => x;
+            }
+        }
+
+        private static readonly Type[] _parseDictionaryParameterTypes = [typeof(IDictionary<string, object?>)];
```

Comment:
Setting this in a static field won't save any computations because static fields in generic classes are unique for different generic arguments. For example, `InputObjectGraphType<Cat>` and `InputObjectGraphType<Dog>` will have different `_parseDictionaryParameterTypes` instances. And since `Initialize` is called only once, it is better to convert it to a local variable to not hold this array in memory for an entire application lifetime.

Reply:
You’re right; I didn’t think of that.